diffsmhm/diff_stats/cuda/sigma.py

```diff
@@ -0,0 +1,408 @@
+import numpy as np
+from numba import cuda
+
+import cmath
+
+
+def _copy_periodic_points_3D(x, y, z, boxsize, buffer_length):
```

Comment:
The appends in this loop are very expensive. Would be better to accumulate the new points in some lists and then append at once.